src/android/HockeyApp.java

```diff
@@ -27,14 +31,67 @@
 import java.lang.Thread;
 import java.text.SimpleDateFormat;
 import java.util.Calendar;
+import java.util.ArrayList;
+
+import android.app.Activity;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.view.View;
 
 public class HockeyApp extends CordovaPlugin {
 
     public static boolean initialized = false;
     public static String appId;
+    public static Object monitor = new Object();
+    public static volatile Bitmap bitmap;
     
     private ConfiguredCrashManagerListener crashListener;
+    
+    // Integration with Crosswalk requires:
+    // - Crosswalk engine version 18 or later
+    // - Latest cordova-plugin-crosswalk-webview plugin     
+    private Bitmap getBitmap() {
+        bitmap = null;
+        boolean isCrosswalk = false;
+        try {
+            Class.forName("org.crosswalk.engine.XWalkWebViewEngine");
+            isCrosswalk = true;
+        } catch (Exception e) {
+        }
+
+        if (isCrosswalk) {
+            long start = System.currentTimeMillis();
+            webView.getPluginManager().postMessage("captureXWalkBitmap", this);
+            try {
+                // Can't take more than 5 seconds to get a screenshot
+                while (bitmap == null && System.currentTimeMillis() - start < 5000) {
+                    synchronized (monitor) {
+                        monitor.wait(50);
+                    }
+                }
+            } catch (InterruptedException e) {
+            }
+        } else {
+            View view = webView.getView();
+            view.setDrawingCacheEnabled(true);
+            bitmap = Bitmap.createBitmap(view.getDrawingCache());
+            view.setDrawingCacheEnabled(false);
+        }
 
+        return bitmap;
+    }
+
+    @Override
+    public Object onMessage(String id, Object data) {
+        if (id.equals("onGotXWalkBitmap")) {
+            bitmap = (Bitmap)data;
```

Comment:
I kept the check in `getBitmap()` as it would be required in that function anyway in case the monitor is interrupted by something other than `onMessage()`